crates/nu-cli/src/repl.rs

```diff
@@ -44,12 +44,31 @@ const PRE_EXECUTE_MARKER: &str = "\x1b]133;C\x1b\\";
 // const CMD_FINISHED_MARKER: &str = "\x1b]133;D;{}\x1b\\";
 const RESET_APPLICATION_MODE: &str = "\x1b[?1l";
 
+// Todo: Find a better place for this or simply make HistorySessionId accessible from the outside
+/// Bypasses the `HistorySessionId` struct not being creatable with a unique value
+/// due to it being private without having to modify the reedline crate.
+/// The unsafe transmute below is also a part of this
```

Comment:
Not sure how this unsafe will fly with the team. I'd try to avoid it if you can.